src/main/java/org/moltimate/moltimatebackend/Structure/StructureUtils.java

```diff
@@ -0,0 +1,184 @@
+package org.moltimate.moltimatebackend.Structure;
+
+import org.biojava.nbio.structure.Atom;
+import org.biojava.nbio.structure.Chain;
+import org.biojava.nbio.structure.Group;
+import org.biojava.nbio.structure.GroupType;
+import org.biojava.nbio.structure.Structure;
+import org.biojava.nbio.structure.geometry.SuperPositions;
+
+import javax.vecmath.Point3d;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StructureUtils {
+
+    /**
+     * Get Residue from a structure
+     *
+     * @param structure:     structure to search for residue in
+     * @param residueName:   three letter name of the residue (ex: ASP)
+     * @param residueNumber: residue number
+     * @return residue if found, null otherwise
+     */
+    public static Group getResidue(Structure structure, String residueName, int residueNumber) {
+        for (Chain chain : structure.getChains()) {
+            for (Group group : chain.getAtomGroups(GroupType.AMINOACID)) {
+                if (group.getChemComp().getThree_letter_code().equals(residueName) && group.getResidueNumber().toString().equals(String.valueOf(residueNumber))) {
```

Comment:
Can you store this boolean expression in a usefully named variable, like `isResidueEqual`, and split the boolean expression up into two lines (start the new line with the `&& group.getResidueNumber()...`

Reply:
Fixed